Runtimes/Core/SwiftOnoneSupport/CMakeLists.txt

```
@@ -0,0 +1,36 @@
+add_library(swiftSwiftOnoneSupport
+  SwiftOnoneSupport.swift
+  "${PROJECT_SOURCE_DIR}/linker-support/magic-symbols-for-install-name.c")
+
+set_target_properties(swiftSwiftOnoneSupport PROPERTIES Swift_MODULE_NAME SwiftOnoneSupport)
+
+if(APPLE AND BUILD_SHARED_LIBS)
+  target_link_options(swiftSwiftOnoneSupport PRIVATE "SHELL:-Xlinker -headerpad_max_install_names")
+endif()
```

I'd like to get @etcwilde's thoughts on this. We _could_ also do this with generator expressions:

```suggestion
target_link_options(swiftSwiftOnoneSupport PRIVATE
  $<$<AND:$<PLATFORM_ID:Darwin>,$<BOOL:${BUILD_SHARED_LIBS}>>:"SHELL:-Xlinker -headerpad_max_install_names>")
```

---

This is the same as what is used for `swiftCore`, so I think it's fine to leave it as-is. If we want the generator expression instead, I think we should change both together.